tests/unit/test_contrib/test_sqlalchemy/test_repository/test_repo_refresh.py

```diff
@@ -0,0 +1,79 @@
+from asyncio import sleep
+from datetime import datetime
+from typing import AsyncGenerator
+
+import pytest
+from sqlalchemy import text
+from sqlalchemy.dialects.mysql import TIMESTAMP
+from sqlalchemy.ext.asyncio import AsyncEngine, AsyncSession, async_sessionmaker
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column
+from sqlalchemy.schema import FetchedValue
+
+from litestar.contrib.sqlalchemy.repository import SQLAlchemyAsyncRepository
+
+pytestmark = [pytest.mark.sqlalchemy_integration]
+
+
+class Base(DeclarativeBase):
+    ...
+
+
+class WithUpdatedField(Base):
+    __tablename__ = "with_updated_field"
+
+    id: Mapped[int] = mapped_column(primary_key=True)
+    val: Mapped[int]
+    updated: Mapped[datetime] = mapped_column(
+        TIMESTAMP, server_default=text("CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP"), server_onupdate=FetchedValue()
+    )
+
+
+@pytest.fixture(name="mysql_engine")
+async def fixture_mysql_engine(asyncmy_engine: AsyncEngine) -> AsyncGenerator[AsyncEngine, None]:
+    async with asyncmy_engine.begin() as conn:
+        await conn.run_sync(WithUpdatedField.__table__.create)  # type:ignore[attr-defined]
+    yield asyncmy_engine
+    async with asyncmy_engine.begin() as conn:
+        await conn.run_sync(WithUpdatedField.__table__.drop)  # type:ignore[attr-defined]
+
+
+@pytest.fixture(name="session")
+async def fixture_session(mysql_engine: AsyncEngine) -> AsyncGenerator[AsyncSession, None]:
+    session = async_sessionmaker(bind=mysql_engine, expire_on_commit=False)()
+    try:
+        yield session
+    finally:
+        await session.rollback()
+        await session.close()
```

Comment:
I think we should run these tests against all of the backends intead of just mysql.

Reply:
I have no problems if you want to do that - I don't know enough about all the other backends to do it myself.

My goal is to have an integration test that guards against future regressions of this on the repository layer so that anyone who tries to remove those `refresh()` calls in the future can see a real world example of why it is there.

Reply:
I've made updates to the test cases and put together a model that uses fetchedvalues on all backends.